Dockerfile

```diff
@@ -13,6 +13,10 @@ COPY api api/
 COPY config config/
 COPY controllers controllers/
 
+# Copy our controller-gen script to work around hermetic build issues
+# See comments in the script itself for more details.
+COPY controller-gen bin/controller-gen-v0.17.2
```

Comment:
When does that version number change? My understanding is that this needs to be kept consistent with go.mod, how do we document that? Also asking because controller-tools release v0.17.3 has been out for three weeks now.

Reply:
Updating it is part of the work we do on each release cycle. It's been kind of a manual process up to now.

With Konflux, we are offered an automated dependency check with PRs that update our go.mod dependencies (that's from Mintmaker), so we might get it like that.
Updating it requires changing both go.mod and Makefile. And now this Dockerfile.

There's no clear documentation about it, maybe there is room for improvement...
At the very least, I can add a comment here.

Reply:
@c3d - I have added a commit (affae46) that tries to remove the need to update all those files with controller-gen version.
Is it overkill?

It can build both on my laptop and in Konflux apparently.